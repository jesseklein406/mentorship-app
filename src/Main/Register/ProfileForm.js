import React from 'react';
import { formValueSelector, reduxForm, Field } from 'redux-form';
import { compose } from 'recompose';
import { connect } from 'react-redux';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import RadioGroup from '../forms/RadioGroup';
import validate from './RegistrationValidation';
import TextField from '../forms/TextField';
import SelectField from '../forms/SelectField';

const experience = [
  { label: '0-1', value: 'entry' },
  { label: '1-3', value: 'junior' },
  { label: '3-7', value: 'intermediate' },
  { label: '7+', value: 'senior' },
];

const interests = [
  { label: 'Portfolio/Code Review', value: 'portfolio' },
  { label: 'Job Search and Interviews', value: 'jobSearch' },
  { label: 'Industry Trends, Skills, Technologies', value: 'skills' },
  { label: 'Leadership, Management', value: 'leadership' },
  { label: 'Business, Entrepreneurship', value: 'business' },
  { label: 'Career Growth', value: 'careerGrowth' },
];

export const ProfileForm = ({ handleSubmit, classes, goToPrevious, accountType = '' }) => (
  <form onSubmit={handleSubmit} noValidate>
    <Typography variant="subheading" color="secondary" gutterBottom>
      CREATE YOUR {accountType.toUpperCase()} PROFILE
    </Typography>
    <Typography variant="subheading" color="secondary" gutterBottom>
      _____
    </Typography>

    <Field name="slackHandle" label="Slack Handle" component={TextField} />
    <Field name="linkedinURL" label="Linkedin URL" component={TextField} />
    <Field name="codeRepoURL" label="Code Repository URL" component={TextField} />
    <Field name="bio" label="Bio" multiline rows="8" component={TextField} />
    <Field name="goals" label="Goals" multiline rows="5" component={TextField} />
    {accountType === 'Mentor' && (
      <Field name="menteeCapacity" label="Mentee Capacity" type="number" component={TextField} />
    )}
    <Field
      name="yearsOfExperience"
      label="Years of Industry Experience"
      component={RadioGroup}
      source={experience}
    />
    <Field name="interests" label="Interests" multiple component={SelectField} items={interests} />

    <Button
      variant="contained"
      color="default"
      size="large"
      className={classes.button}
      onClick={goToPrevious}
    >
      BACK
    </Button>
    <Button variant="raised" type="submit" color="primary" size="large" className={classes.button}>
      CONTINUE
    </Button>
  </form>
);

ProfileForm.displayName = 'ProfileForm';

const selector = formValueSelector('registration');

const mapStateToProps = state => ({
  accountType: selector(state, 'accountType'),
});

export default compose(
  connect(mapStateToProps),
  reduxForm({
    form: 'registration',
    destroyOnUnmount: false,
    forceUnregisterOnUnmount: true,
    validate,
    initialValues: {
      interests: [],
    },
    onSubmit: () => {
      console.log('not yet implemented');
    },
  })
)(ProfileForm);
