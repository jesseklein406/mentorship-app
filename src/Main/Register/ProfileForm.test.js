import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Button from '@material-ui/core/Button';

import { ProfileForm } from './ProfileForm';

const defaultProps = {
  classes: {},
  goToPrevious: jest.fn(),
};

const setup = props => shallow(<ProfileForm {...defaultProps} {...props} />);

describe('<ProfileForm />', () => {
  it('renders snapshot correctly', () => {
    const wrapper = setup();

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the correct title when accountType = Mentor', () => {
    const wrapper = setup({ accountType: 'Mentor' });

    expect(wrapper.findWhere(n => n.text() === 'MENTEE')).toHaveLength(0);
    expect(wrapper.findWhere(n => n.text() === 'MENTOR')).toHaveLength(1);
  });

  it('renders the correct title when accountType = Mentee', () => {
    const wrapper = setup({ accountType: 'Mentee' });

    expect(wrapper.findWhere(n => n.text() === 'MENTEE')).toHaveLength(1);
    expect(wrapper.findWhere(n => n.text() === 'MENTOR')).toHaveLength(0);
  });

  it('renders mentee capacity when accountType = Mentor', () => {
    const wrapper = setup({ accountType: 'Mentor' });

    expect(wrapper.find('[name="menteeCapacity"]')).toHaveLength(1);
  });

  it('does not render mentee capacity when accountType = Mentee', () => {
    const wrapper = setup({ accountType: 'Mentee' });

    expect(wrapper.find('[name="menteeCapacity"]')).toHaveLength(0);
  });

  it('clicking the back button calls the goToPrevious prop', () => {
    const wrapper = setup();
    wrapper
      .find(Button)
      .first()
      .simulate('click');

    expect(defaultProps.goToPrevious).toHaveBeenCalledTimes(1);
  });
});
