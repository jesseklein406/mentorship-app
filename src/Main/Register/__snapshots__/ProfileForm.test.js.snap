// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileForm /> renders snapshot correctly 1`] = `
<form
  noValidate={true}
>
  <WithStyles(Typography)
    color="secondary"
    gutterBottom={true}
    variant="subheading"
  >
    CREATE YOUR 
     PROFILE
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="secondary"
    gutterBottom={true}
    variant="subheading"
  >
    _____
  </WithStyles(Typography)>
  <Field
    component={[Function]}
    label="Slack Handle"
    name="slackHandle"
  />
  <Field
    component={[Function]}
    label="Linkedin URL"
    name="linkedinURL"
  />
  <Field
    component={[Function]}
    label="Code Repository URL"
    name="codeRepoURL"
  />
  <Field
    component={[Function]}
    label="Bio"
    multiline={true}
    name="bio"
    rows="8"
  />
  <Field
    component={[Function]}
    label="Goals"
    multiline={true}
    name="goals"
    rows="5"
  />
  <Field
    component={[Function]}
    label="Years of Industry Experience"
    name="yearsOfExperience"
    source={
      Array [
        Object {
          "label": "0-1",
          "value": "entry",
        },
        Object {
          "label": "1-3",
          "value": "junior",
        },
        Object {
          "label": "3-7",
          "value": "intermediate",
        },
        Object {
          "label": "7+",
          "value": "senior",
        },
      ]
    }
  />
  <Field
    component={[Function]}
    items={
      Array [
        Object {
          "label": "Portfolio/Code Review",
          "value": "portfolio",
        },
        Object {
          "label": "Job Search and Interviews",
          "value": "jobSearch",
        },
        Object {
          "label": "Industry Trends, Skills, Technologies",
          "value": "skills",
        },
        Object {
          "label": "Leadership, Management",
          "value": "leadership",
        },
        Object {
          "label": "Business, Entrepreneurship",
          "value": "business",
        },
        Object {
          "label": "Career Growth",
          "value": "careerGrowth",
        },
      ]
    }
    label="Interests"
    multiple={true}
    name="interests"
  />
  <WithStyles(Button)
    color="default"
    onClick={[Function]}
    size="large"
    variant="contained"
  >
    BACK
  </WithStyles(Button)>
  <WithStyles(Button)
    color="primary"
    size="large"
    type="submit"
    variant="raised"
  >
    CONTINUE
  </WithStyles(Button)>
</form>
`;
